### PR TITLE
Log a warning when log file is rotated before tailer finishes 

### PR DESCRIPTION
**Pull request by gh123man**
### What does this PR do?
Log a warning when a log file is rotated but has not finished tailing the file. 
Made the rotation stop timeout configurable. 

### Describe your test plan
Added UT for configurable timeout
manually tested warning logic

### CODE REVIEW

File: pkg/logs/input/file/tailer.go [hunk 1]
```diff
@@ -124,14 +124,20 @@ func (t *Tailer) Start(offset int64, whence int) error {
...
 	for {
+		n, err := t.read()
+		if err != nil {
+			return
+		}
+
 		select {
 		case <-t.stop:
+			if n != 0 && t.didFileRotate == 1 {
+				log.Warn("Tailer stopped after rotation close timeout with data remaining")
```
Comment (gh123man) [inline pkg/logs/input/file/tailer.go:135]: Any suggestion for better wording here?

Comment (truthbk) [reply pkg/logs/input/file/tailer.go:135]:
```suggestion
				log.Warn("Tailer stopped after rotation close timeout with remaining unread data")
```

Comment (truthbk) [reply pkg/logs/input/file/tailer.go:135]: Perhaps? Not really sure it's any better.

**Review by truthbk — APPROVED** [review]
This is looking good to me, no objections to making the close timeout period configurable.

**Review by olivielpeau — APPROVED** [review]

**Pull request merged**